File: api/app/api/v1/endpoints/media.py
from fastapi import APIRouter, Depends, HTTPException
from app.agents.video_orchestrator import VideoOrchestrator
from app.agents.audio_orchestrator import AudioOrchestrator
from app.api.v1.endpoints.auth import get_current_user
from app.domain.models.media import (
    StartVideoConversationRequest, 
    VideoConversationResponse,
    StartAudioConversationRequest,
    AudioConversationConfigResponse
)

router = APIRouter()

@router.post("/video/start_conversation", response_model=VideoConversationResponse)
async def start_video_conversation(
    request: StartVideoConversationRequest,
    orchestrator: VideoOrchestrator = Depends(VideoOrchestrator),
    user: dict = Depends(get_current_user)
):
    try:
        response = await orchestrator.start_conversation(request)
        return response
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.post("/audio/start_conversation", response_model=AudioConversationConfigResponse)
def get_audio_conversation_config(
    request: StartAudioConversationRequest,
    orchestrator: AudioOrchestrator = Depends(AudioOrchestrator),
    user: dict = Depends(get_current_user)
):
    try:
        response = orchestrator.generate_config(request)
        return response
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e)) 

File: www/lib/api/media.ts
import { apiClient } from './auth';

export interface StartVideoConversationRequest {
  permit_details: Record<string, any>;
}

export interface VideoConversationResponse {
  conversation_url: string;
}

export interface StartAudioConversationRequest {
  permit_details: Record<string, any>;
}

export interface AudioConversationConfigResponse {
  agent_id: string;
  prompt: string;
}

class MediaApi {
  async startVideoConversation(request: StartVideoConversationRequest): Promise<VideoConversationResponse> {
    try {
      const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/api/v1/media/video/start_conversation`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${apiClient.getAccessToken()}`,
        },
        body: JSON.stringify(request),
      });

      if (!response.ok) {
        const error = await response.json();
        throw new Error(error.detail || 'Failed to start video conversation');
      }

      return response.json();
    } catch (error) {
      console.error('Video conversation failed:', error);
      throw new Error(error instanceof Error ? error.message : 'Failed to start video conversation');
    }
  }

  async startAudioConversation(request: StartAudioConversationRequest): Promise<AudioConversationConfigResponse> {
    try {
      const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/api/v1/media/audio/start_conversation`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${apiClient.getAccessToken()}`,
        },
        body: JSON.stringify(request),
      });

      if (!response.ok) {
        const error = await response.json();
        throw new Error(error.detail || 'Failed to start audio conversation');
      }

      return response.json();
    } catch (error) {
      console.error('Audio conversation failed:', error);
      throw new Error(error instanceof Error ? error.message : 'Failed to start audio conversation');
    }
  }
}

export const mediaApi = new MediaApi(); 